src/lib/list/list.ts

```diff
@@ -50,6 +61,334 @@ export class MdList {
+  @Input() checkboxPosition: 'before' | 'after' = 'after';
+
+  /** Whether the option is disabled. */
+  @Input()
+  get disabled() { return (this.selectionList && this.selectionList.disabled) || this._disabled; }
+  set disabled(value: any) { this._disabled = coerceBooleanProperty(value); }
+
+  @Input()
+  get value() { return this._value; }
+  set value( val: any) { this._value = coerceBooleanProperty(val); }
+
+  @Input()
+  get selected() { return this._selected; }
+  set selected( val: boolean) { this._selected = coerceBooleanProperty(val); }
+
+  /** Emitted when the option is focused. */
+  onFocus = new EventEmitter<MdSelectionListOptionEvent>();
+
+  /** Emitted when the option is selected. */
+  @Output() select = new EventEmitter<MdSelectionListOptionEvent>();
+
+  /** Emitted when the option is deselected. */
+  @Output() deselect = new EventEmitter<MdSelectionListOptionEvent>();
+
+  /** Emitted when the option is destroyed. */
+  @Output() destroy = new EventEmitter<MdSelectionListOptionEvent>();
+
+  constructor(private _renderer: Renderer2,
+              private _element: ElementRef,
+              private _changeDetector: ChangeDetectorRef,
+              @Optional() public selectionList: MdSelectionList,) { }
+
+
+  ngAfterContentInit() {
+    this._lineSetter = new MdLineSetter(this._lines, this._renderer, this._element);
+  }
+
+  ngOnDestroy(): void {
+    this.destroy.emit({option: this});
+  }
+
+  toggle(): void {
+    if(this._disabled == false) {
+      this.selected = !this.selected;
+      this.selectionList.selectedOptions.toggle(this);
+      this._changeDetector.markForCheck();
+    }
+  }
+
+  /** Allows for programmatic focusing of the option. */
+  focus(): void {
+    this._element.nativeElement.focus();
+    this.onFocus.emit({option: this});
+  }
+
+  /** Whether this list item should show a ripple effect when clicked.  */
+  isRippleEnabled() {
+    return !this.disableRipple && !this.selectionList.disableRipple;
+  }
+
+  _handleFocus() {
+    this._hasFocus = true;
+    this._renderer.addClass(this._element.nativeElement, FOCUSED_STYLE);
+  }
+
+  _handleBlur() {
+    this._renderer.removeClass(this._element.nativeElement, FOCUSED_STYLE);
+  }
+
+  /** Retrieves the DOM element of the component host. */
+  _getHostElement(): HTMLElement {
+    return this._element.nativeElement;
+  }
+}
+
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-selection-list, mat-selection-list',
+  host: {
+    'role': 'listbox',
+    '[attr.tabindex]': '_tabIndex',
+    'class': 'mat-selection-list',
+    '(focus)': 'focus()',
+    '(keydown)': 'keydown($event)'},
+  queries: {
+    options: new ContentChildren(MdListOption)
+  },
+  template: '<ng-content></ng-content>',
+  styleUrls: ['list.css'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MdSelectionList implements AfterContentInit, OnDestroy {
```

Comment:
Can `md-selection-list` and `md-list-option` go in separate files? (`selection-list.ts` and `list-option.ts`)